pages/index.tsx

```diff
@@ -1,19 +1,160 @@
+import Head from "next/head";
+import { useEffect, useState } from "react";
+import styles from "../styles/Home.module.css";
+import type {
+  Applicant,
+  ApplicantDisplay,
+  ScreenerOptions,
+} from "./api/lib/applicant";
+import debounce from "debounce";
+
+const debouncedGetSearchResults = debounce(
+  async (
+    name: string,
+    sortColumn: string,
+    sortOrder: string,
+    filter: string,
+    page: number,
+    pageSize: number,
+    setApplicants: (applicants: Applicant[], total: number) => void
+  ) => {
+    let response = await fetch("/api/applicants", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name,
+        sortColumn,
+        sortOrder,
+        filter,
+        page,
+        pageSize,
+      }),
+    });
+
+    const awaitedResponse = await response.json();
+    setApplicants(awaitedResponse.applicants, awaitedResponse.total);
+  },
+  1000
+);
+
+const columns: (keyof ApplicantDisplay)[] = ["name", "phone", "screener"];
+const filterOptions: ("all" | ScreenerOptions)[] = [
+  "all",
+  "pending",
+  "approved",
+  "rejected",
+];
 
 const Home: NextPage = () => {
-  const [applicants, setApplicants] = useState<Applicant[]>([])
+  const [loading, setLoading] = useState(true);
+  const [applicants, setApplicants] = useState<Applicant[]>([]);
+  const [totalApplicants, setTotalApplicants] = useState<number>(0);
+  const [search, setSearch] = useState<string>("");
+  const [filter, setFilter] = useState<string>("");
+  const [sortColumn, setSortColumn] = useState<string>("name");
+  const [sortOrder, setSortOrder] = useState<string>("asc");
+  const [page, setPage] = useState<number>(1);
+  const [pageSize] = useState<number>(10);
 
   useEffect(() => {
     (async () => {
-      setApplicants(await (await fetch('/api/all')).json() as Applicant[]);
+      const response = await fetch("/api/applicants", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          name: search,
+          sortColumn,
+          sortOrder,
+          filter,
+          page,
+          pageSize,
+        }),
+      });
+      const data = await response.json();
+      setApplicants(data.applicants);
+      setTotalApplicants(data.total);
+      setLoading(false)
     })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const handleChange = (string: string) => {
+    setSearch(string);
+    setPage(1);
+    debouncedGetSearchResults(
+      string,
+      sortColumn,
+      sortOrder,
+      filter,
+      1,
+      pageSize,
+      (applicants, total) => {
+        setApplicants(applicants);
+        setTotalApplicants(total);
+      }
+    );
+  };
+
+  const handleSort = (
+    column: keyof ApplicantDisplay,
+    order: "asc" | "desc"
+  ) => {
+    setSortColumn(column);
+    setSortOrder(order);
+    setPage(1);
```

Comment:
Before this, it was possible to get to a page that should have been disabled. If the user filtered by a screener, went to the next page, and then filtered by a different screener, the page would stay the same. So if there was no second page of results for the new filter, they would see a blank table.